core/src/test/java/org/wso2/carbon/kernel/transports/TransportManagerTest.java

```diff
@@ -118,4 +118,49 @@ public void testSuccessfulBeginMaintenance() {
         Assert.assertTrue(true);
     }
 
+    @Test(dependsOnMethods = {"testSuccessfulBeginMaintenance"})
+    public void testSuccessfulEndMaintenance() {
+        try {
+            transportManager.endMaintenance();
+            transportManager.unregisterTransport(carbonTransport2);
+        } catch (IllegalStateException e) {
+            Assert.assertTrue(false);
+        }
```

Comment:
Again this code is obsolete. Now the line 127 is as follows;

Assert.assertTrue(false, "attempting to unregister transport in an illegal state");

If an exception is thrown, then the test should fail. But if it is not thrown, then it passes. So in order for this test to pass the exception should not get thrown, which is what happen now.